python/eos/plot/plotter.py

```diff
@@ -105,18 +105,21 @@ def setup_plot(self):
             self.ax.xaxis.set_minor_locator(matplotlib.ticker.AutoMinorLocator())
             self.ax.xaxis.set_ticks_position('both')
 
-            if 'format' in myx:
-                self.xformat = myx['format']
-            else:
-                self.xformat = '${x}$'
-
             if 'scale' in myx:
+
+                if 'format' in myx:
+                    self.xformat = myx['format']
+                else:
+                    self.xformat = '${x}$'
+                    eos.warn("Argument plot:x:format might be required when using plot:x:scale to avoid side effects")
+
                 self.xscale = float(myx['scale'])
                 self.xticks = matplotlib.ticker.FuncFormatter(lambda x, pos, xscale=self.xscale: self.xformat.format(x=x / xscale))
-            else:
-                self.xticks = matplotlib.ticker.StrMethodFormatter(self.xformat)
+                self.ax.xaxis.set_major_formatter(self.xticks)
 
-            self.ax.xaxis.set_major_formatter(self.xticks)
+            else:
+                if 'format' in myx:
+                    eos.warn("Argument plot:x:format is only used when plot:x:scale is used")
```

Comment:
This change should also be reflected in the documentation (``doc/api/python.rst``)

Reply:
Done!